components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -253,7 +253,12 @@ private void validateRequestAttributes(User recoveredUser, RecoveryScenarios sce
     private void triggerNotification(User user, String notificationChannel, String templateName, String code,
                                      String eventName, Property[] metaProperties) throws IdentityRecoveryException {
 
+        String serviceProviderUUID = (String) IdentityUtil.threadLocalProperties.get().get(IdentityEventConstants.EventProperty.SERVICE_PROVIDER_UUID);
+
         HashMap<String, Object> properties = new HashMap<>();
+        if (serviceProviderUUID != null && !serviceProviderUUID.isEmpty()) {
```

Comment:
Can we use isBlank() instead? Check other places for the same thing.

Reply:
ack

Reply:
this was revert back as isblank() throw null pointer exception for null value.